### DISCUSSION TITLE
length command for strings

### DISCUSSION THREAD

**Discussion by morzel85** [nushell/nushell#8004] [General]
Shouldn't `length` command reject arguments that are not a list or record, or (if Nushell allows it) reuse the implementation of `str length`?

For strings it does not throw an error but returns rather misleading result:

![image](https://user-images.githubusercontent.com/7968512/217593981-c0b67912-8912-4747-b875-5b1b2c2a19a2.png)

**Comment by fdncred** [comment]
i've done that before too. maybe if you give it a string it should suggest str length. i can see length working on list, record, table, not sure what else.

**Comment by morzel85** [comment]
If this is the signature of `length`:

```
  list<any> | length -> <int>
  <table> | length -> <int>
```
then why isn't Nu rejecting `string` argument? Does it apply some type coercion?
